nexus/src/authn/external/session_cookie.rs

```diff
@@ -22,24 +23,31 @@ pub trait Session {
     fn time_created(&self) -> DateTime<Utc>;
 }
 
+pub trait SessionToken: From<String> + Clone + Display {}
+impl<T: From<String> + Clone + Display> SessionToken for T {}
+
```

Comment:
lol at this pattern I found online. Is this really the way to do this?

Reply:
I think this is a common pattern for defining a sort of trait alias.  Dropshot uses it too.

This is different than creating a newtype, though.  If the goal is to prevent people from accidentally using some random String as a SessionToken, I think you'd want a separate newtype for it.  I _think_ even with this change, you could provide a `String` or a `&str` to one of the session methods even for the `SessionStore` impl on `Arc<ServerContext>`.  Is that what you want?  (What exactly is the goal here?)

Reply:
I'm not sure there is any point to it, since at least for now there are no conditions on turning a string into a `SessionToken` and I can't think of any we'd want. I had this on my to-do list as a followup to the session work, but I'm perfectly happy to scrap this is if doesn't seem necessary.

Reply:
I feel like a newtype might be useful similar to `Name`: if we have a `Name`, we know it's been validated and we can't accidentally confuse a regular String for one that's been validated.